### PR TITLE
Fix using SignTool options

### PR DESCRIPTION
**Pull request by YehudaKremer**
Hello @zobo 

In this PR I'm trying to fix problems with SignTool.

this PR includes the changes: 

1. user can use SignTool freely as the office [documentation](https://learn.microsoft.com/en-us/dotnet/framework/tools/signtool-exe) instructing
2. when trying to sign with an installed certificate by using one of the options '**/n**', '**/i**' and '**/sha1**', this package will try to extract the certificate **Subject** with PowerShell script [you provided](https://github.com/YehudaKremer/msix/pull/155#issuecomment-1290140779)

About [.crt](https://github.com/YehudaKremer/msix/pull/155#issuecomment-1290151151), I don't think SignTool supports .crt files, if I am wrong can you please provide me an example, thanks.

I would appreciate it if you could go through the [changes](https://github.com/YehudaKremer/msix/blob/1dc68f2baeb7b89f917cb96a77a04672b60dd12b/lib/src/sign_tool.dart#L19) in your spare time 👍

### CODE REVIEW

File: lib/src/sign_tool.dart [hunk 1]
```diff
@@ -15,44 +15,138 @@ class SignTool {
...
+  void _failToGetCertificateSubject() {
+    _logger.stdout('could not find the Publisher value.'.red);
+    _logger.stdout(
+        'you must provide the Publisher value at "msix_config: publisher" in the pubspec.yaml file'
+            .red);
+    _logger.stdout(
+        'the Publisher is the certificate "Subject" in this exact format: "CN=Contoso Software, O=Contoso Corporation, C=US"');
+    _logger.stdout('see where you can found your certificate Subject:');
+    _logger.stdout(
+        'https://drive.google.com/file/d/1oAsnrp2Kf-jZ_kaRjyF5llQ0YZy1IwNe/view?usp=sharing'
```
Comment (zobo) [inline lib/src/sign_tool.dart:59]: I suggest you store the image into this GitHub repository and link to it via a commit url - to make it permanent.

**Comment by zobo** [comment]
The PR looks good, indeed the simple _split by space_ caused issues when I was experimenting. I like the use of `extension`s.

Regarding CRT. The docs say:
> /f SignCertFile
> Specifies the signing certificate in a file. If the file is in Personal Information Exchange (PFX) format and protected by a password, use the /p option to specify the password. **If the file does not contain private keys**, use the /csp and /kc options to specify the CSP and private key container name.

My experience is that if I provide a `.crt` file, the SignTool will figure out what private key to use. I did not need to add `/csp` and `/kc` options.

I'd suggest to also look for the case where a certificate file is provided, but no password and then extract the subject via:
`new-object System.Security.Cryptography.X509Certificates.X509Certificate2("..\file.crt") | select -expandproperty Subject -First 1`

I'll try to provide the code, but I'll be leaving for a week of travel soon.

File: lib/src/sign_tool.dart [hunk 2]
```diff
@@ -15,44 +15,138 @@ class SignTool {
...
+        subject = await _getCertificateSubjectByThumbprint();
+      } else if (_config.signToolOptions!.containsArguments(['/n', '/r'])) {
+        subject = await _getCertificateSubjectBySubject();
+      } else if (_config.signToolOptions!.containsArgument('/i')) {
+        subject = await _getCertificateSubjectByIssuer();
+      }
+    } else if (_config.certificatePath != null &&
+        extension(_config.certificatePath!).toLowerCase() == '.pfx') {
+      subject = await _getPfxCertificateSubject();
+    }
```
Comment (zobo) [inline lib/src/sign_tool.dart:35]: Quick dry code - have not tested it....
```
else if (_config.certificatePath != null)  {
  subject = await _getCrtCertificateSubject();
}
```


```
  Future<String> _getCrtCertificateSubject() async {
    _logger.trace('getting crt certificate Subject');

    var certificateDetailsProcess = await _executePowershellCommand(
        "new-object System.Security.Cryptography.X509Certificates.X509Certificate2(\"${_config.certificatePath}\") | select -expandproperty Subject -First 1
");

    certificateDetailsProcess.exitOnError();

    var subject = (certificateDetailsProcess.stdout as String).trim();

    return subject;
  }
```

**Comment by YehudaKremer** [comment]
Good notes and examples,
I will add them to this branch soon as possible.

Thank You! 👍

File: lib/src/sign_tool.dart (hunk 2 above)
Comment (YehudaKremer) [reply lib/src/sign_tool.dart:37]: done here: https://github.com/YehudaKremer/msix/pull/156/commits/b252a98818abbd2b2341525aa283c2e9e02af00b

File: lib/src/sign_tool.dart (hunk 1 above)
Comment (YehudaKremer) [reply lib/src/sign_tool.dart]: done here https://github.com/YehudaKremer/msix/pull/156/commits/114944835b5b57dcecac0a87541941f7c1650632

File: lib/src/sign_tool.dart (hunk 2 above)
Comment (YehudaKremer) [reply lib/src/sign_tool.dart:37]: also added the documentation https://github.com/YehudaKremer/msix/pull/156/commits/bc0d17ef64e3d35e9018a0d085e23b9ad9368974

**Comment by YehudaKremer** [comment]
Hello @zobo 

I publish a new version (3.7.0) with these changes: 

- https://github.com/YehudaKremer/msix/releases/tag/3.7.0
- https://pub.dev/packages/msix/versions/3.7.0

Thank you!

**Comment by zobo** [comment]
Looks like everything is working great. Thank you again!

**Pull request merged**